source/MRViewer/MRRenderMeshObject.cpp

```diff
@@ -1020,6 +1120,43 @@ RenderBufferRef<uint8_t> RenderMeshObject::loadTexturePerFaceTextureBuffer_()
     return buffer;
 }
 
+RenderBufferRef<MR::Vector3f> RenderMeshObject::loadPointPosBuffer_()
+{
+    auto& glBuffer = GLStaticHolder::getStaticGLBuffer();
+    if ( !dirtyPointPos_ || !objMesh_->mesh() )
+        return glBuffer.prepareBuffer<Vector3f>( pointSize_, false );
+
+    MR_NAMED_TIMER( "mesh_points_dirty_positions" );
+
+    const auto& mesh = objMesh_->mesh();
+    const auto& topology = mesh->topology;
+
+    auto buffer = glBuffer.prepareBuffer<Vector3f>( pointSize_ = topology.lastValidVert() + 1 );
+    std::copy( MR::begin( mesh->points ), MR::begin( mesh->points ) + pointSize_, buffer.data() );
+    return buffer;
+}
+
+
+RenderBufferRef<VertId> RenderMeshObject::loadPointValidIndicesBuffer_()
+{
+    auto& glBuffer = GLStaticHolder::getStaticGLBuffer();
+    if ( !dirtyPointPos_ || !objMesh_->mesh() )
+        return glBuffer.prepareBuffer<VertId>( pointValidSize_, !pointValidBuffer_.valid() );
+
+    MR_NAMED_TIMER( "mesh_points_dirty_valid_indices" );
+
+    const auto& validPoints = objMesh_->mesh()->topology.getValidVerts();
+    const size_t lastValid = validPoints.find_last() + 1;
+    pointValidSize_ = int( validPoints.count() );
+    auto buffer = glBuffer.prepareBuffer<VertId>( pointValidSize_ );
+    for ( int in = 0, out = 0; in < lastValid; ++in )
+    {
+        if ( validPoints.test( VertId( in ) ) )
+            buffer[out++] = VertId( in );
+    }
+    return buffer;
```

Comment:
it shoud respect corner rendering  (as far as we will use vertices from mesh buffer)

Reply:
reworked